app/controllers/api/v1/migrations/external_controller.rb

```diff
@@ -38,17 +42,19 @@ def create_role
         # Does: Creates a user.
 
         def create_user
-          language = if user_params[:language].blank? || user_params[:language] == 'default'
-                       I18n.default_locale
-                     else
-                       user_params[:language]
-                     end
+          user_hash = user_params.to_h
+
+          return render_data status: :created if User.exists? email: user_hash[:email], provider: 'greenlight'
```

Comment:
This is an approach where we make an prefight query to check wither or not the resource has been created,
This has the following advantages:
1. It's simpler.
2. Whenever a resource exists it will fail in one query => faster duplicated request check + no trigger of additional logic (other queries mainly).
But it suffers from the following disadvantages:
1. For the initial/original requests this caused an additional query.